### PR TITLE
Add cdf and icdf methods for StudentT distribution

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Implement `cdf` and `icdf` methods for the StudentT distribution. This enables quantile computation for the StudentT `DistributionForecast` (e.g., for `SimpleFeedForwardEstimator`).
- The new `cdf` and `icdf` methods use `scipy.stats.t` under the hood, so  require moving data to CPU and are not differentiable.
- Move `StudentTDistribtuionOutput` to `gluonts.torch.distributions.studentT`.
- Add a test that ensures that torch and scipy versions have the same parametrization.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** enhancement

### CODE REVIEW

**Comment by kashif** [comment]
I could try to add the appropriate methods to pytorch?

**Comment by kashif** [comment]
ah damn no `betainc` in pytorch i believe...

**Comment by shchur** [comment]
> ah damn no `betainc` in pytorch i believe...

Yeah, pytorch doesn't have the necessary special functions, unfortunately

**Comment by lostella** [comment]
> ah damn no `betainc` in pytorch i believe...

Sounds like that needs to be added too 🙃

**Comment by kashif** [comment]
tensorflow uses the implementation from `eigen` i forget if that is available on the pytorch side...

File: src/gluonts/torch/distributions/studentT.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        df = self.df.detach().cpu().numpy()
+        loc = self.loc.detach().cpu().numpy()
+        scale = self.scale.detach().cpu().numpy()
+        self.scipy_student_t = ScipyStudentT(df=df, loc=loc, scale=scale)
+
+    def cdf(self, value: torch.Tensor) -> torch.Tensor:
+        if self._validate_args:
+            self._validate_sample(value)
+        result = self.scipy_student_t.cdf(value.detach().cpu().numpy())
+        return torch.tensor(result, device=value.device, dtype=value.dtype)
```
Comment (kashif) [inline src/gluonts/torch/distributions/studentT.py:50]: shall we also `require_grads=False` this tensor and the one below?

Comment (shchur) [reply src/gluonts/torch/distributions/studentT.py:50]: I thought this was already the [default](https://pytorch.org/docs/stable/generated/torch.tensor.html), are there some cases where the output ends up having `requires_grad=True`?

Comment (kashif) [reply src/gluonts/torch/distributions/studentT.py:50]: you are right! my bad

**Review by lostella — APPROVED** [review]
Awesome 👏🏻 thanks @shchur!
